pkg/eval/builtin_fn_container.go

```diff
@@ -896,7 +899,58 @@ const (
 	uncomparable
 )
 
-func compare(a, b interface{}) ordering {
+//elvdoc:fn compare
+//
+// ```elvish
+// compare $a $b
+// ```
+//
+// Outputs -1 if `$a` is less than `$b`, 0 if equal, and 1 if greater than.
+//
+// The following comparison algorithm is used:
```

Comment:
This algorithm is now documented in both `order` and `compare`; better to avoid the duplication.